pkg/cmd/step/e2e/step_e2e_gc.go

```diff
@@ -63,12 +70,61 @@ func NewCmdStepE2EGC(commonOpts *opts.CommonOptions) *cobra.Command {
 	cmd.Flags().StringVarP(&options.Region, "region", "", "europe-west1-c", "GKE region to use. Default: europe-west1-c")
 	cmd.Flags().StringVarP(&options.ProjectID, "project-id", "p", "", "Google Project ID to delete cluster from")
 	cmd.Flags().IntVarP(&options.Duration, "duration", "d", 2, "How many hours old a cluster should be before it is deleted if it does not have a --delete tag")
+	cmd.Flags().StringArrayVarP(&options.Providers, "providers", "", []string{"gke"}, "The providers to run the cleanup for")
 
 	return cmd
 }
 
 // Run runs the command
 func (o *StepE2EGCOptions) Run() error {
+	// Until https://github.com/jenkins-x/jx/issues/6206 is done, we are going to be using different approaches to run this for different providers
+	for _, pr := range o.Providers {
+		switch strings.ToLower(pr) {
+		case cloud.GKE:
+			return o.gcpGarbageCollection()
+		case cloud.AWS:
+			fallthrough
+		case cloud.EKS:
+			return o.eksGarbageCollection()
+		default:
+			return fmt.Errorf("provider %s doesn't have an E2E GC implementation defined", pr)
+		}
+	}
+	return nil
+}
+
+func (o *StepE2EGCOptions) eksGarbageCollection() error {
+	eksClient, err := factory.NewClientForProvider(cloud.EKS)
+	if err != nil {
+		return errors.Wrap(err, "could not obtain an EKS cluster client to ")
+	}
+	eksClusters, err := eksClient.List()
+	if err != nil {
+		return errors.Wrap(err, "there was a problem obtaining every eksClient in the current account")
+	}
+
+	for _, eksCluster := range eksClusters {
+		if eksCluster.Status == "ACTIVE" {
+			if !o.ShouldDeleteMarkedEKSCluster(eksCluster) {
```

Comment:
The negation does somehow not compute in my mind. `ShouldDeleteMarkedEKSCluster` implies to me that if returns true in the case the cluster should be deleted, but the opposite seems to be the case. I think the function is named incorrectly